src/core/compact_object.cc

```diff
@@ -469,8 +469,8 @@ int RobjWrapper::ZsetAdd(double score, sds ele, int in_flags, int* out_flags, do
 
       /* check if the element is too large or the list
        * becomes too long *before* executing zzlInsert. */
-      if (zl_len + 1 > server.zset_max_listpack_entries ||
-          sdslen(ele) > server.zset_max_listpack_value || !lpSafeToAdd(lp, sdslen(ele))) {
+      if (zl_len >= server.zset_max_listpack_entries ||
```

Comment:
lpSafeToAdd checks if sdslen is not huge but we already check it with `sdslen(ele) > server.zset_max_listpack_value`